src/orion/algo/pbt/pbt.py

```diff
@@ -0,0 +1,877 @@
+        trials at `population_size`.  For efficiency it is better to have less workers running than
+        total population_size. Default: 50.
+    generations: int, optional
+        Number of generations, from lowest fidelity to highest one. This will determine how
+        many branchings occur during the execution of PBT. Default: 10
+    exploit: dict or None, optional
+        In the mutate part, one can define the customized mutate function with its mutate factors,
+        such as multiply factor (times/divides by a multiply factor) and add factor
+        (add/subtract by a multiply factor). The function must be defined by
+        an importable string. If None, default
+        mutate function is used: ``orion.algo.mutate_functions.default_mutate``.
+    explore: dict or None, optional
+        In the mutate part, one can define the customized mutate function with its mutate factors,
+        such as multiply factor (times/divides by a multiply factor) and add factor
+        (add/subtract by a multiply factor). The function must be defined by
+        an importable string. If None, default
+        mutate function is used: ``orion.algo.mutate_functions.default_mutate``.
+    fork_timeout: int, optional
+        Maximum amount of time in seconds that an attempt to mutate a trial should take, otherwise
+        algorithm.suggest() will raise ``SuggestionTimeout``. Default: 60
+
+    """
+
+    requires_type = None
+    requires_dist = "linear"
+    requires_shape = "flattened"
+
+    def __init__(
+        self,
+        space,
+        seed=None,
+        population_size=50,
+        generations=10,
+        exploit=None,
+        explore=None,
+        fork_timeout=60,
+    ):
+        if exploit is None:
+            exploit = {
+                "of_type": "PipelineExploit",
+                "exploit_configs": [
+                    {
+                        "of_type": "BacktrackExploit",
+                        "min_forking_population": 5,
+                        "truncation_quantile": 0.9,
+                        "candidate_pool_ratio": 0.2,
+                    },
+                    {
+                        "of_type": "TruncateExploit",
+                        "min_forking_population": 5,
+                        "truncation_quantile": 0.8,
+                        "candidate_pool_ratio": 0.2,
+                    },
+                ],
+            }
+
+        if explore is None:
+            explore = {
+                "of_type": "PipelineExplore",
+                "explore_configs": [
+                    {"of_type": "ResampleExplore", "probability": 0.2},
+                    {"of_type": "PerturbExplore", "factor": 1.2, "volatility": 0.0001},
+                ],
+            }
+
+        self.random_search = Random(space)
+        self._queue = []
+
+        fidelity_index = self.fidelity_index
+        if fidelity_index is None:
+            raise RuntimeError(SPACE_ERROR)
```

Comment:
where is this `SPACE_ERROR` defined?

Reply:
Oops, not defined indeed!

Reply:
Fixed in https://github.com/Epistimio/orion/pull/705/commits/753d8d2fc5e494835a9aefc802e4e9d1e717009e